frontend/src/metabase/setup/actions.ts

```diff
@@ -223,6 +223,38 @@ export const SUBMIT_SETUP = "metabase/setup/SUBMIT_SETUP";
 export const submitSetup = createAsyncThunk<void, void, ThunkConfig>(
   SUBMIT_SETUP,
   async (_, { dispatch }) => {
+    dispatch(setEmbeddingHomepageFlags());
     dispatch(goToNextStep());
   },
 );
+
+export const setEmbeddingHomepageFlags = createAsyncThunk(
+  "setup/setEmbeddingHomepageFlags",
+  async (_, { getState, dispatch }) => {
+    const usageReason = getUsageReason(getState());
+    const tokenFeatures = getSetting(getState(), "token-features");
+    const adminSettings = getSettings(getState());
+    const enableEmbeddingSetByEnv = adminSettings.find(
+      (setting: { key: string }) => setting.key === "enable-embedding",
+    )?.is_env_setting;
+
+    const interestedInEmbedding =
+      usageReason === "embedding" || usageReason === "both";
+    const isLicenseActive = tokenFeatures && tokenFeatures["embedding"];
```

Comment:
[slack thread](https://metaboat.slack.com/archives/C063Q3F1HPF/p1711128542316109) where I asked if we should check specifically for this feature, meanwhile I've optimistically changed the code to that as I think that's what we should do